### ISSUE TITLE
Isn't <p> suitable as a View root element?

### ISSUE THREAD

**Issue by maiorfi** [yyx990803/vue#700]
Hi. I just wrote a simple demo where a table contained some v-repeat'ed `<tr>`.
Such a table was contained into a `<p>` chosen as View root element for a Vue viewmodel.

That way, it didn't work (templates weren't replaced). Changing `<p>` into a `<div>` solved the issue.

Same thing happens if, referring to GitHub commits fiddler sample, you change root `<div>` element into a `<p>`.

Is it a bug or the expected behavior?

**Comment by TMiguelT** [comment]
Can you edit your post and put backticks (`) around your HTML so that the markdown parser doesn't delete it all?

**Comment by maiorfi** [comment]
Done, thanks.

**Comment by TMiguelT** [comment]
Thanks, much easier to read! 
According to the HTML spec, [a `<p>` element](https://developer.mozilla.org/en/docs/Web/HTML/Element/p#Summary) can only contain ['phrasing content'](https://developer.mozilla.org/en-US/docs/Web/Guide/HTML/Content_categories#Phrasing_content), i.e. content that is mostly text and related markup. A table is not in the list of accepted child nodes, so it's quite possible your browser deleted your component because it was placing a `<table>` inside a `<p>`. I've had issues like this in the past. 

So basically yes this is expected behaviour but it's not a Vue issue, it's related to the HTML spec. The obvious solution is to not use `<p>` elements as the direct root of Vue components.
